app/components/MyComponent.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  Text,
+  View,
+  TouchableOpacity,
+  FlatList,
+  TextInput,
+} from "react-native";
+
+const MyComponent = ({ data }) => {
+  const [selectedItems, setSelectedItems] = useState([]);
+  const [dataSource, setDataSource] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    setDataSource(data);
+  }, [data]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setDataSource(data.filter((item) => item.name.includes(searchTerm)));
+    }, 1000);
+  }, [searchTerm]);
```

Comment:
no need for this side effect as we can directly create filtered data from `data`

_we can do case insensitive search for filtering data_


```suggestion
  const filteredData =  data.filter((item) => item.name.toLowerCase().includes(searchTerm.toLowerCase()));
```